astropy/coordinates/representation.py

```diff
@@ -2006,13 +2006,22 @@ def _get_deriv_key(self, base):
             d_comp = getattr(self, 'd_{0}'.format(name), None)
             if d_comp is not None:
                 d_unit = comp.unit / d_comp.unit
-                # Get the si unit without a scale by going via Quantity;
-                # `.si` causes the scale to be included in the value.
-                return str(u.Quantity(1., d_unit).si.unit)
+
+                # This is quite a bit faster than using to_system() or going
+                # through Quantity()
+                d_unit_si = d_unit.decompose(u.si.bases)
+                d_unit_si._scale = 1 # remove the scale from the unit
```

Comment:
This worries me a bit because I'm not 100% sure some of this stuff doesn't get cached?  But eh, we can live dangerously.

Reply:
I can't wait until we rediscover this comment in 17 months after spending a day debugging some obscure caching bug!